python/mxnet/gluon/rnn/rnn_layer.py

```diff
@@ -173,67 +177,63 @@ def begin_state(self, batch_size=0, func=ndarray.zeros, **kwargs):
             states.append(func(name='%sh0_%d'%(self.prefix, i), **info))
         return states
 
-    def forward(self, inputs, states=None):
-        batch_size = inputs.shape[self._layout.find('N')]
+    def hybrid_forward(self, F, inputs, states=None, **kwargs):
+        if F is ndarray:
+            batch_size = inputs.shape[self._layout.find('N')]
+            if self._input_size == 0:
```

Comment:
implement infershape instead?

Reply:
this is not possible due to the inverse shape inference in concat.

Reply:
or did you mean overriding block's infer shape? I'm taking a look